### ISSUE TITLE
binary hinge loss

### ISSUE THREAD

**Issue by ng0323** [Lasagne/Lasagne#744]
Hi,
In the code for binary hinge loss, the targets are normalized to {-1,1}. 
Shouldn't the same apply for the predictions if they are in (0,1)? I mean, the predictions should also be normalized to (-1,1).

**Comment by f0k** [comment]
Oh. You're right, it doesn't make sense the way it's documented.
For reference, the hinge loss is `max(0, delta - predictions * targets)`, with delta defaulting to 1.

If the predictions are sigmoid in (0,1), then for a negative target, the loss will always be between delta and delta + 1. That's not intended.
If the predictions are tanh in (-1,1), then for the default delta of 1, the `max(0, ...)` does not make sense. That's also not necessarily intended.
To be equivalent to an SVM, you'd actually feed the log-odds (before applying the sigmoid): https://en.wikipedia.org/wiki/Hinge_loss

I think rather than changing the implementation, we should change the documentation to encourage feeding the log-odds (sometimes called logits).

**Comment by ng0323** [comment]
Ok, thanks for the clarification.
My predictions are from softmax, so I suppose  I should take its log-odds.

**Comment by f0k** [comment]
(Reopened because the documentation still needs fixing.)

**Comment by f0k** [comment]
Okay, I've adapted the documentation accordingly. We could also compute the log-odds from the sigmoid output (as log(y / (1-y))), but this requires a Theano optimization to get rid of the sigmoid afterwards. I propose to add this optimization to Theano, then add a `log_odds` argument to define whether the predictions are log-odds or probabilities. This would have to wait until 0.3.

Thinking again, we should add the `log_odds` argument right away so we can change its default to `False` in 0.3 and warn about this upcoming change now. This way it will not silently change behavior in 0.2, and become a drop-in replacement for binary cross-entropy in 0.3. PR in #771.
